lib/ansible/modules/windows/win_certificate_info.ps1

```diff
@@ -0,0 +1,145 @@
+    $cert_info.issued_to = $cert.GetNameInfo('SimpleName', $false)
+    $cert_info.signature_algorithm = $cert.SignatureAlgorithm.FriendlyName
+    $cert_info.dns_names = @($cert_info.issued_to)
+    $cert_info.raw = [System.Convert]::ToBase64String($cert.GetRawCertData())
+    $cert_info.public_key = [System.Convert]::ToBase64String($cert.GetPublicKey())
+    $cert_info.extensions = @()
+    foreach ($extension in $cert.Extensions)
+    {
+        $cert_info.extensions += @{
+            critical = $extension.Critical
+            field    = $extension.Oid.FriendlyName
+            value    = $extension.Format($false)
+        }
+        if ($extension -is [System.Security.Cryptography.X509Certificates.X509BasicConstraintsExtension])
+        {
+            $cert_info.is_ca = $extension.CertificateAuthority
+            $cert_info.path_length_constraint = $extension.PathLengthConstraint
+        }
+        elseif ($extension -is [System.Security.Cryptography.X509Certificates.X509EnhancedKeyUsageExtension])
+        {
+            $cert_info.intended_purposes = $extension.EnhancedKeyUsages.FriendlyName -as [string[]]
+        }
+        elseif ($extension -is [System.Security.Cryptography.X509Certificates.X509KeyUsageExtension])
+        {
+            $cert_info.key_usages = $extension.KeyUsages.ToString().Split(',').Trim() -as [string[]]
+        }
+        elseif ($extension -is [System.Security.Cryptography.X509Certificates.X509SubjectKeyIdentifierExtension])
+        {
+            $cert_info.ski = $extension.SubjectKeyIdentifier
+        }
+        elseif ($extension.Oid.value -eq '2.5.29.17')
+        {
+            $sans = $extension.Format($true).Split("`r`n", [System.StringSplitOptions]::RemoveEmptyEntries)
+            foreach ($san in $sans)
+            {
+                $san_parts = $san.Split("=")
+                if ($san_parts.Length -ge 2 -and $san_parts[0].Trim() -eq 'DNS Name')
+                {
+                    $cert_info.dns_names += $san_parts[1].Trim()
+                }
+            }
+        }
+    }
+    return $cert_info
+}
+
+$store_location_values = ([System.Security.Cryptography.X509Certificates.StoreLocation]).GetEnumValues() | ForEach-Object { $_.ToString() }
+
+$spec = @{
+    options = @{
+        thumbprint     = @{ type = "str"; required = $false }
+        store_name     = @{ type = "str"; default = "My"; }
+        store_location = @{ type = "str"; default = "LocalMachine"; choices = $store_location_values; }
+    }
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$thumbprint = $module.Params.thumbprint
+$store_name = $module.Params.store_name
+$store_location = [System.Security.Cryptography.X509Certificates.Storelocation]"$($module.Params.store_location)"
+
+$store = New-Object -TypeName System.Security.Cryptography.X509Certificates.X509Store -ArgumentList $store_name, $store_location
+try
+{
+    $store.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadWrite)
```

Comment:
Is ReadWrite needed, I would have assumed `Read` is what we need for this exercise.